basic_bootloader/src/bootloader/run_single_interaction.rs

```diff
@@ -55,6 +64,22 @@ where
                 }
             })?;
 
+        // Emit mint event
```

Comment:
I don't like it and it would be very nice to not emit this event. Basically this is completely unexpected by EVM - this is more or less okay I guess, since it only happens in our specific type of transactions (l1->l2, upgrade).